client/src/pages/email-sign-up.tsx

```diff
@@ -61,24 +67,11 @@ function AcceptPrivacyTerms({
     acceptTermsRef.current = acceptTerms;
   });
 
-  useEffect(() => {
-    return () => {
-      // if a user navigates away from here we should set acceptedPrivacyTerms
-      // to true (so they do not get pulled back) without changing their email
-      // preferences (hence the null payload)
-      // This makes sure that the user has to opt in to Quincy's emails and that
-      // they are only asked twice
-      if (!acceptedPrivacyRef.current) {
-        acceptTermsRef.current(null);
-      }
-    };
-  }, []);
-
```

Comment:
We'll still need this to set the `acceptedPrivacyTerms` property when they leave this page. Otherwise they'll get redirected here the next time they log in.

We don't care about redirecting users away from here if they choose to navigate to the page and we want to allow people to move away (to settings, in this case). With that in mind I recommend stopping those two types of redirection and, instead, just send the user to learn after they click yes or no to Quincy's newsletter.

It's a bit fiddly to explain, but I'm happy to push a commit if that sounds reasonable.

Reply:
Trouble is we don't know what percentage it is or how likely we are to lose people if we keep bouncing them back to this page every time they go to /learn. It's intensely frustrating, so I really don't think that's the desired UX.

I can see an argument for sending people back here after they log out and back in again, but not trapping them on this page.

> The key is a misnomer and might as well be renamed to isOnboardingDone

Agreed. Another thing we should migrate!

Also @ahmadabdolsaheb what do you think about adding analytics to the three 'exits' i.e. yes to emails, no to emails and everything else.

Reply:
All I'm saying is that the UX of

1. clicks on freecodecamp banner
2. gets bounced back to /email-signup/
3. clicks on menu > curriculum
4. gets bounced back to /email-signup/

is a rubbish UX pattern and if a site did that to me, I probably wouldn't return.

> In any case, I am not in favor of the effect hack we had, and unless you are proposing some workaround like that, I am happy to look at your code, and you can post it here - so we can keep this moving.

Depends what you want, but I'll push the code that prevents what I discussed above and we can see where to go next.

Reply:
I know what you are saying; you do not need to repeat yourself :) - but you are looking over the fact that you may be overthinking the scenario.

> but I'll push the code that prevents what I discussed above and we can see where to go next.

In that case, I will wait for you to make a separate PR instead. I have this in my ever-going backlog, which Quincy requested, and I do not want to hold the feature over a nitpick.

You are more than welcome to refine the UX as you like later.

Reply:
It wasn't a tricky fix, so I'd already pushed before checking my notifications :shrug:

Reply:
No worries!